### ISSUE TITLE
Online Co-op: Swapping keys in Kitten Chamber rooms causes weird issues

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#488]
Reported in 0.1.3.

>i threw and picked up the silver and gold keys and used the silver key normally, then i picked up the gold key, but the gold key remained attached to me in the same position as if it were being held...
switching weapons, attacking, sheilding, using the warp, and dying with revival did nothing to unstick the key from me
>
>the key was also unable to unlock the door and couldn't be interacted with in any way

**Comment by ChrisNZL** [comment]
Additionally, throwing keys very fast online (like against a wall over and over again) is causing issues with being unable to use the key on any door.

`NetworkMessage_InteractWithDungeonRoomObject`

Server is calling this warning:

```csharp
if (!objectStruct.CanBeActivatedByCreatureStruct(creatureStruct)) {
	#if HEADLESS_SERVER
		Debug.LogWarning("WARNING: objectStruct "+objectNetID+" can't be activated by creature "+creatureNetID);
	#endif
	return;
}
```

**Comment by ChrisNZL** [comment]
Fixed in 0.1.6.

Refactored many NetworkMessages to cater for having the server tell clients to throw, drop, or consume keys as needed:
- `NetworkMessage_DealDamageToCreature`
- `NetworkMessage_InteractWithDungeonRoomObject`
- `NetworkMessage_MakeCreatureDrinkPotion`
- `NetworkMessage_MakeCreatureEquipItem`
- `NetworkMessage_PickUpTemporaryItem`
- `NetworkMessage_PlayerDisconnected`
- `NetworkMessage_SetCreatureAttackState`
- `NetworkMessage_SetShieldState`

Minor known issue:
- Clients can't pick up the same key type (eg if holding a yellow key and trying to pick-up a second yellow key), but this is not currently possible with how keys are spawned, and no errors are generated – just something to consider fixing in the future if multiple yellow keys can spawn in a room later on.
